casper/src/main/scala/coop/rchain/casper/engine/LastFinalizedStateBlockRequester.scala

```diff
@@ -112,10 +116,19 @@ object LastFinalizedStateBlockRequester {
         // Mark block as received
         isReceived <- Stream.eval(d.modify(_.received(block.blockHash)))
 
+        allValidatorsSeen <- Stream.eval(validatorsRequired.modify[Boolean] { currSet =>
+                              val newSet = currSet - block.sender
+                              (newSet, newSet.isEmpty)
+                            })
+
+        inDeployLifeSpanRage = ProtoUtil.blockNumber(block) > ProtoUtil.blockNumber(
+          approvedBlock.candidate.block
+          // TODO extract this number from shard config
+        ) - 50
```

Comment:
why 50? I remember we got a config for this. But I can not find it.

Reply:
Its still hardcoded n Casper implementation, we are going to make it consistent <https://github.com/rchain/rchain/pull/2986>